src/Xamarin.Android.Build.Tasks/Tasks/AppendCustomMetadataToItemGroup.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks {
+	public class AppendCustomMetadataToItemGroup : Task {
+		[Required]
+		public ITaskItem[] Inputs { get; set; }
+
+		[Required]
+		public ITaskItem[] MetaDataItems { get; set; }
+
+		[Output]
+		public ITaskItem[] Output { get; set; }
+
+		public override bool Execute ()
+		{
+			var output = new List<ITaskItem> ();
+
+			foreach (var item in Inputs) {
+				var fn = Path.GetFileNameWithoutExtension (item.ItemSpec);
```

Comment:
@Redth, would we care about paths or extensions? I am not sure on this.

Reply:
Paths could be different on each users machine. So we need to match on the Filename only, we could include the extension if need be. However our existing SkipCases item group only uses the filename see https://github.com/xamarin/xamarin-android/blob/bad41a64f91768d63a6068d2cc4dab2073128514/src/Xamarin.Android.Build.Tasks/Xamarin.Android.SkipCases.projitems#L19